libcontainer/user/user.go

```diff
@@ -473,7 +473,7 @@ func GetAdditionalGroups(additionalGroups []string, group io.Reader) ([]int, err
 				return nil, fmt.Errorf("Unable to find group %s", ag)
 			}
 			// Ensure gid is inside gid range.
-			if gid < minId || gid > maxId {
```

Comment:
Are you sure this change is correct?

Reply:
Well, it works and passes tests on amd64 and i386 as far as I can tell. I think it is correct. Can we get someone to review?

Reply:
It's not so much that it passes tests, it's that you're now making it impossible for someone to have an additional group that is equal to `maxId` and I was wondering whether this you think it's actually semantically correct (I'm not entirely sure -- though I will double-check).

(I am the one doing the review. :wink:)

Reply:
This is very related to failure and I _had to_ change `>` to `>=` because of the following test failure on _amd64_ introduced as direct result of integer overflow fix:

~~~~
--- FAIL: TestGetAdditionalGroups (0.00s)
        user_test.go:459: Parse(user.foo{groups:[]string{"2147483647"}, expected:[]int(nil), hasError:true}) expects error but has none
~~~~

That looks perfectly reasonable to be but may be therei is a better way.

Reply:
Ah okay. I'll take another look at whether or not this is semantically okay and get back to you.